src/segment_six.c

```diff
@@ -68,58 +73,79 @@ static void hour_display_update_proc(Layer *layer, GContext* ctx) {
   }
 }
 
-static void handle_minute_tick(struct tm *tick_time, TimeUnits units_changed) {
+static void prv_mark_dirty() {
   layer_mark_dirty(s_minute_display_layer);
   layer_mark_dirty(s_hour_display_layer);
 }
 
-static void main_window_load(Window *window) {
+static void prv_handle_minute_tick(struct tm *tick_time, TimeUnits units_changed) {
+  prv_mark_dirty();
+}
+
+// Event fires frequently, while obstruction is appearing or disappearing
+static void prv_unobstructed_change(AnimationProgress progress, void *context) {
+  prv_mark_dirty();
+}
+
+// Event fires once, after obstruction appears or disappears
+static void prv_unobstructed_did_change(void *context) {
+  prv_mark_dirty();
+}
+
+static void prv_main_window_load(Window *window) {
   Layer *window_layer = window_get_root_layer(window);
-  GRect bounds = layer_get_bounds(window_layer);
+  GRect bounds = layer_get_unobstructed_bounds(window_layer);
```

Comment:
I believe this is correct. The window background color is already set on init(), then these layers are above that.